plugin-lifecycle/src/main/java/com/mapbox/maps/plugin/lifecycle/MapboxLifecyclePluginImpl.kt

```diff
@@ -0,0 +1,50 @@
+package com.mapbox.maps.plugin.lifecycle
+
+import android.widget.FrameLayout
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleObserver
+import androidx.lifecycle.OnLifecycleEvent
+import androidx.lifecycle.ViewTreeLifecycleOwner
+import com.mapbox.maps.MapboxLifecycleObserver
+import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
+
+/**
+ * Concrete implementation of MapboxLifecyclePlugin.
+ */
+class MapboxLifecyclePluginImpl : MapboxLifecyclePlugin {
+  /**
+   * Register a MapboxLifecycleObserver to observe life cycle events from LifecycleOwner
+   *
+   * @param mapView the instance of mapView, will get the LifecycleOwner from mapview's parent
+   * @param observer the observer that listen to the life cycle events
+   */
+  override fun registerLifecycleObserver(mapView: FrameLayout, observer: MapboxLifecycleObserver) {
+    ViewTreeLifecycleOwner.get(mapView)?.apply {
```

Comment:
What if the `ViewTreeLifecycleOwner.get(mapView)` returns null? shall we log some error messages or exceptions to notify the developer?

Reply:
The screen would be black. Yeah, we could log the error message.